From 31a7dc36b357d423e6e381d5cf29fdc7b4fee659 Mon Sep 17 00:00:00 2001
From: Marian Chereji <marian.chereji@freescale.com>
Date: Tue, 7 Aug 2012 19:52:50 +0000
Subject: [PATCH 271/518] dpa_classifier: Bug fix. Avoid dereferencing NULL
 pointer

Added a fix to remove danger of reading NULL pnext pointer.

Signed-off-by: Marian Chereji <marian.chereji@freescale.com>
[Grabbed from the branch, LINUX_IR5.2.0, of
https://git.freescale.com/git-private/cgit.cgi/ppc/alu-b4860/linux.git.]
Signed-off-by: Tiejun Chen <tiejun.chen@windriver.com>
---
 drivers/staging/fsl_dpa_offload/dpa_classifier.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/staging/fsl_dpa_offload/dpa_classifier.c b/drivers/staging/fsl_dpa_offload/dpa_classifier.c
index b48f422..e301fea 100644
--- a/drivers/staging/fsl_dpa_offload/dpa_classifier.c
+++ b/drivers/staging/fsl_dpa_offload/dpa_classifier.c
@@ -2963,7 +2963,7 @@ static int init_nat_hm(struct dpa_cls_hm *pnat_hm)
 	 * Update the low level ops chain head for all the other
 	 * high level ops in the high level ops chain:
 	 */
-	if (pnat_hm->hm_chain != pnext_hm->hm_chain) {
+	if ((pnext_hm) && (pnat_hm->hm_chain != pnext_hm->hm_chain)) {
 		list_for_each_entry(pcurrent,
 			&pnat_hm->list_node, list_node) {
 			pcurrent->hm_chain = pnat_hm->hm_chain;
-- 
1.7.5.4

